web/client/components/share/SharePanel.jsx

```diff
@@ -151,6 +151,9 @@ class SharePanel extends React.Component {
             !isEqual(this.props.isVisible, newProps.isVisible)) {
             this.initializeDefaults(newProps);
         }
+        if (!isEqual(this.props?.viewerOptions?.orientation, newProps.viewerOptions?.orientation)) {
+            this.initializeDefaults(newProps);
+        }
```

Comment:
Why not have this block here with another `||` operator https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/share/SharePanel.jsx#L151? We could avoid a separate if block.